modules/stochastic_tools/src/reporters/SolutionContainer.C

```diff
@@ -15,60 +15,33 @@ registerMooseObject("StochasticToolsApp", SolutionContainer);
 InputParameters
 SolutionContainer::validParams()
 {
-  InputParameters params = GeneralReporter::validParams();
+  InputParameters params = SnapshotContainerBase::validParams();
   params.addClassDescription(
       "Class responsible for collecting distributed solution vectors into a container. We append "
       "a new distributed solution vector (containing all variables) at every execution.");
   MooseEnum system_type("nonlinear aux", "nonlinear");
   params.addParam<MooseEnum>(
       "system", system_type, "The system whose solution should be collected.");
-  params.addParam<NonlinearSystemName>(
-      "nonlinear_system_name",
-      "nl0",
-      "Option to select which nonlinear system's solution shall be stored.");
+
   return params;
 }
 
 SolutionContainer::SolutionContainer(const InputParameters & parameters)
-  : GeneralReporter(parameters),
-    _accumulated_solutions(
-        declareRestartableData<std::vector<std::unique_ptr<NumericVector<Number>>>>(
-            "accumulated_solution")),
-    _system_type(getParam<MooseEnum>("system")),
-    _nonlinear_system_number(
-        _fe_problem.nlSysNum(getParam<NonlinearSystemName>("nonlinear_system_name")))
+  : SnapshotContainerBase(parameters), _system_type(getParam<MooseEnum>("system"))
```

Comment:
Could you add an error (that I forgot back in the day) if an aux system is selected and the `nonlinear_system_name` is defined by the user? You could use `isParamSetByUser` I think.